# HG changeset patch
# User Rik <rik@octave.org>
# Date 1462224472 25200
#      Mon May 02 14:27:52 2016 -0700
# Node ID 79414f8e6f24a7f927db6a252b6e045173c09a4a
# Parent  5b9868c2e212b3e01fb308c3656e01548ea67beb
schur.cc: Declare selector variable as volatile to suppress gcc warning.

schur.cc (init): Declare selector variable as volatile to suppress gcc warning.

diff --git a/liboctave/numeric/schur.cc b/liboctave/numeric/schur.cc
--- a/liboctave/numeric/schur.cc
+++ b/liboctave/numeric/schur.cc
@@ -168,23 +168,21 @@ schur<Matrix>::init (const Matrix& a, co
   else
     jobvs = 'N';
 
   char ord_char = ord.empty () ? 'U' : ord[0];
 
   if (ord_char == 'A' || ord_char == 'D' || ord_char == 'a' || ord_char == 'd')
     sort = 'S';
 
-  double_selector selector = 0;
+  volatile double_selector selector = 0;
   if (ord_char == 'A' || ord_char == 'a')
     selector = select_ana<double>;
   else if (ord_char == 'D' || ord_char == 'd')
     selector = select_dig<double>;
-  else
-    selector = 0;
 
   octave_idx_type n = a_nc;
   octave_idx_type lwork = 8 * n;
   octave_idx_type liwork = 1;
   octave_idx_type info;
   octave_idx_type sdim;
   double rconde;
   double rcondv;
@@ -257,23 +255,21 @@ schur<FloatMatrix>::init (const FloatMat
   else
     jobvs = 'N';
 
   char ord_char = ord.empty () ? 'U' : ord[0];
 
   if (ord_char == 'A' || ord_char == 'D' || ord_char == 'a' || ord_char == 'd')
     sort = 'S';
 
-  float_selector selector = 0;
+  volatile float_selector selector = 0;
   if (ord_char == 'A' || ord_char == 'a')
     selector = select_ana<float>;
   else if (ord_char == 'D' || ord_char == 'd')
     selector = select_dig<float>;
-  else
-    selector = 0;
 
   octave_idx_type n = a_nc;
   octave_idx_type lwork = 8 * n;
   octave_idx_type liwork = 1;
   octave_idx_type info;
   octave_idx_type sdim;
   float rconde;
   float rcondv;
@@ -346,23 +342,21 @@ schur<ComplexMatrix>::init (const Comple
   else
     jobvs = 'N';
 
   char ord_char = ord.empty () ? 'U' : ord[0];
 
   if (ord_char == 'A' || ord_char == 'D' || ord_char == 'a' || ord_char == 'd')
     sort = 'S';
 
-  complex_selector selector = 0;
+  volatile complex_selector selector = 0;
   if (ord_char == 'A' || ord_char == 'a')
     selector = select_ana<Complex>;
   else if (ord_char == 'D' || ord_char == 'd')
     selector = select_dig<Complex>;
-  else
-    selector = 0;
 
   octave_idx_type n = a_nc;
   octave_idx_type lwork = 8 * n;
   octave_idx_type info;
   octave_idx_type sdim;
   double rconde;
   double rcondv;
 
@@ -455,23 +449,21 @@ schur<FloatComplexMatrix>::init (const F
   else
     jobvs = 'N';
 
   char ord_char = ord.empty () ? 'U' : ord[0];
 
   if (ord_char == 'A' || ord_char == 'D' || ord_char == 'a' || ord_char == 'd')
     sort = 'S';
 
-  float_complex_selector selector = 0;
+  volatile float_complex_selector selector = 0;
   if (ord_char == 'A' || ord_char == 'a')
     selector = select_ana<FloatComplex>;
   else if (ord_char == 'D' || ord_char == 'd')
     selector = select_dig<FloatComplex>;
-  else
-    selector = 0;
 
   octave_idx_type n = a_nc;
   octave_idx_type lwork = 8 * n;
   octave_idx_type info;
   octave_idx_type sdim;
   float rconde;
   float rcondv;
 
